test/remote/gateways/remote_simetrik_test.rb

```diff
@@ -0,0 +1,427 @@
+      trace_id: @authorize_capture_options_success[:trace_id]
+    }
+
+    assert capture = @gateway.capture(@amount, auth.authorization, option)
+    assert_failure capture
+    assert_equal 'CAPTURE_REJECTED', capture.params['message']
+  end
+
+  def test_successful_void
+    auth = @gateway.authorize(@amount, @credit_card, @authorize_void_options_success)
+    assert_success auth
+
+    option = {
+      token_acquirer: @token_acquirer,
+      trace_id: @authorize_capture_options_success[:trace_id],
+      acquire_extra_options: {}
+    }
+    sleep(3)
+    assert void = @gateway.void(auth.authorization, option)
+    assert_success void
+    assert_equal 'successful void', void.params['message']
+  end
+
+  def test_failed_void
+    # First successful void
+    auth = @gateway.authorize(@amount, @credit_card, @authorize_void_options_success)
+    assert_success auth
+
+    option = {
+      token_acquirer: @token_acquirer,
+      trace_id: @authorize_capture_options_success[:trace_id],
+      acquire_extra_options: {}
+    }
+    sleep(3)
+    assert void = @gateway.void(auth.authorization, option)
+    assert_success void
+    assert_equal 'successful void', void.params['message']
+
+    # Second failed void
+    option = {
+      token_acquirer: @token_acquirer,
+      trace_id: '2717a3e0-0db2-4971-b94f-686d3b72c44b'
+    }
+    void = @gateway.void(auth.authorization, option)
+    assert_failure void
+    assert_equal 'VOID_REJECTED', void.params['message']
+  end
+
+  def test_successful_refund
+    option = {
+      token_acquirer: 'bd4c0f26-a357-4294-9b9e-a90e6c868c6e',
+      trace_id: '23138123-321213',
+      comment: 'Daniel Bernal',
+      currency: 'PEN',
+      acquire_extra_options: {
+        ruc: '20202380621'
+      }
+    }
+
+    assert refund = @gateway.refund(200, '936ca6a48a8b4cd3a3050bc637a5e2bc', option)
+    assert_success refund
+    assert_equal 'successful refund', refund.message
+  end
+
+  def test_failed_refund
+    response = @gateway.purchase(@amount, @credit_card, @authorize_capture_options_success)
+    option = {
+      token_acquirer: @token_acquirer,
+      trace_id: '2717a3e0-0db2-4971-b94f-686d3b72c44b',
+      currency: 'USD',
+      comment: 'This is a comment',
+      acquire_extra_options: {
+        ruc: '13431131234'
+      }
+    }
+    sleep(3)
+    refund = @gateway.refund(@amount, response.authorization, option)
+    assert_failure refund
+    assert_equal('REFUND_REJECTED', refund.params['message'])
+  end
+
+  def test_transcript_scrubbing
+    transcript = capture_transcript(@gateway) do
+      @gateway.purchase(@amount, @credit_card, @authorize_capture_options_success)
+    end
+    transcript = @gateway.scrub(transcript)
+    assert_scrubbed(@credit_card.number, transcript)
+    assert_scrubbed(@credit_card.verification_value.to_s, transcript)
+  end
+
+  private
+
+  def setup_options
+    setup_authorize_void_success_options()
+    @authorize_capture_options_failed = {
```

Comment:
Two new test cases have been added: `test_failed_purchase_by_invalid_card` and `test_failed_authorize_by_invalid_card`, these two contain all the correct data but are rejected because an invalid card is used.